docs/docsite/rst/dev_guidelines.rst

```diff
@@ -481,19 +481,128 @@ should return in the module.  As well as information related to the call itself,
 some response metadata.  It is OK to return this to the user as well as they may find it useful.
 
 Boto3 returns all values CamelCased.  Ansible follows Python standards for variable names and uses
-snake_case. There is a helper function in module_utils/ec2.py called ``camel_dict_to_snake_dict``
+snake_case.  There is a helper function in module_utils/ec2.py called ``camel_dict_to_snake_dict``
 that allows you to easily convert the boto3 response to snake_case.
 
 You should use this helper function and avoid changing the names of values returned by Boto3.
 E.g. if boto3 returns a value called 'SecretAccessKey' do not change it to 'AccessKey'.
 
+There is an optional parameter, ``ignore_list``, which is used to avoid converting a sub-tree
+of a dict.  This is particularly important for tags, where keys are case-sensitive.  We
+convert the 'Tags' key but nothing below.
+
+.. code-block:: python
+
+   # Make a call to AWS
+   result = connection.aws_call()
+
+   # Return the result to the user without modifying tags
+   module.exit_json(changed=True, **camel_dict_to_snake_dict(result, ignore_list=['Tags']))
+
+Tags
+----
+
+Tags should be returned as a dictionary of key: value pairs, with each key being the tag's
+key and value being the tag's value.  It should be noted, however, that boto3 returns tags
+as a list of dictionaries.
```

Comment:
```suggestion
key and value being the tag's value.  It should be noted, however, that boto3 often returns tags
as a list of dictionaries.
```

With my recent tagging cleanup I've found a number of the newer APIs have started returning dicts rather than lists of dicts.